### ISSUE TITLE
`react-web-api/no-leaked-event-listener` should not flag eventListener with `{ signal }` options

### ISSUE THREAD

**Issue by SukkaW** [Rel1cx/eslint-react#1622]
### Problem Description

If an `addEventListener` was provided with `{ signal }` option:

```ts
window.addEventListener('resize', () => {
  echartInstanceRef.current?.resize();
}, { signal });
```

Then `react-web-api/no-leaked-event-listener` should not flag this usage, as `AbortSignal` here clearly indicates there would be some kind of clean up.

### Alternative Solutions

N/A

### Feature Details

`react-web-api/no-leaked-event-listener` should not flag `addEventListener` with `{ signal }` options. Do note that the `signal` should not be nullish.

### Examples

```tsx
import { useEffect } from 'foxact/use-abortable-effect';

function Comp() {
  useEffect(signal => {
    window.addEventListener('resize', () => console.log('whoa!'), { signal });
  }, []);
}
```

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users
